server/backend/api/app/crud/services/permission_audit_service.py

```diff
@@ -0,0 +1,172 @@
+        return PrivilegeDetailsSchema(
+            permission_type=PrivilegeDetailsPermissionTypeEnum.END_USER,
+            roles=[
+                PrivilegeDetailsRoleSchema(
+                    role=enduser_privliege_list[0].detail.role.display_name,
+                    scopes=scopes
+                )
+            ]
+        )
+
+    @staticmethod
+    def to_enduser_privliege_revoked_details(delete_record: FamUserRoleXref):
+        # Note, current FAM supports deleting ONLY 1 role with ONLY 1 scope type ("CLIENT") with 1 forst client.
+        # TODO !! When team begins new scoped roles model, it needs refactoring.  # noqa NOSONAR
+        revoked_permission_role = delete_record.role
+        is_forest_client_scoped_role = revoked_permission_role.client_number_id
+        forest_client_number = None
+        forest_client_name = None
+        # Search forest client name for storing audit record. Current FAM does not store forest client name in db.
+        if (is_forest_client_scoped_role):
+            forest_client_number = revoked_permission_role.client_number.forest_client_number
+            api_instance_env = crud_utils.use_api_instance_by_app(revoked_permission_role.application)
+            forest_client_integration_service = ForestClientIntegrationService(api_instance_env)
+            fc_search_result = forest_client_integration_service.find_by_client_number(
+                forest_client_number
+            )
+            # Forest Client search is an exact search.
+            if len(fc_search_result) == 1:
+                forest_client_name = fc_search_result[0]["clientName"]
+            else:
+                error_msg = (
+                    "Revoke user permission encountered problem."
+                    + f"Unknown forest client number {forest_client_number} for "
+                    + "scoped permission {revoked_permission_role.role_name}."
+                )
+                LOGGER.debug(error_msg)
+                raise HTTPException(
+                    status_code=HTTPStatus.INTERNAL_SERVER_ERROR,
+                    detail={
+                        "code": ERROR_CODE_UNKNOWN_STATE,
+                        "description": error_msg
+                    }
+                )
+
+        revoke_privilege_details_scope = PrivilegeDetailsScopeSchema(
+            scope_type=PrivilegeDetailsScopeTypeEnum.CLIENT,
+            client_id=forest_client_number if is_forest_client_scoped_role else None,
+            client_name=forest_client_name if is_forest_client_scoped_role else None
+        )
+
+        return PrivilegeDetailsSchema(
+            permission_type=PrivilegeDetailsPermissionTypeEnum.END_USER,
+            roles=[
+                PrivilegeDetailsRoleSchema(
+                    role=revoked_permission_role.display_name,
+                    scopes=[revoke_privilege_details_scope]
+                )
+            ]
+        )
```

Comment:
This is only convenient method for transformation between schema objects to build the end user `enduser_privliege_revoked_details`